tests/app/main/test_services.py

```diff
@@ -544,6 +544,32 @@ def test_should_view_public_service_with_update_message(self, data_api_client):
             res.get_data(as_text=True)
         )
 
+    def test_editable_empty_questions_are_shown(self, data_api_client):
+        """Currently the only question that is editable but potentially empty (optional during service submission)
+        is the serviceDefinitionDocumentURL for G-Cloud 9 so we test this behaviour in this class"""
+        service_kwargs = {
+            "serviceName": "My G-Cloud 9 service",
+            "lot": "cloud-hosting",
+            "lotName": "Cloud Hosting",
+            "lotSlug": "cloud-hosting"
+        }
+        service_kwargs.update(self.framework_kwargs)
+        self.login()
+        self._setup_service(
+            data_api_client,
+            service_status='published',
+            **service_kwargs
+        )
+
+        res = self.client.get("/suppliers/frameworks/{}/services/123".format(self.framework_kwargs["framework_slug"]))
+
+        assert res.status_code == 200
+        # There's no serviceDefinitionDocumentURL in the test service data, but the row should still be shown
+        self.assert_in_strip_whitespace(
+            'Service definition document',
+            res.get_data(as_text=True)
+        )
```

Comment:
Should we assert the empty message itself, as well as the label?

Reply:
Yes, probably!